src/services/expensesService.js

```diff
@@ -0,0 +1,72 @@
+const { Op } = require('sequelize');
+const { models } = require('../models/models.js');
+const { Expense } = models;
+
+const getAllExpenses = async (userId, categories, from, to) => {
+  const where = {};
+
+  if (userId) {
+    where.userId = userId;
+  }
+
+  if (categories) {
+    where.category = { [Op.in]: categories };
+  }
+
+  if (from) {
+    where.spentAt = { [Op.gte]: new Date(from) };
+  }
+
+  if (to) {
+    where.spentAt = {
+      ...where.spentAt,
+      [Op.lte]: new Date(to),
```

Comment:
Ensure that the `to` date is properly validated before using it to prevent potential errors with invalid date formats.